core/dbt/clients/jinja.py

```diff
@@ -493,13 +493,28 @@ def _requote_result(raw_value: str, rendered: str) -> str:
     return f'{quote_char}{rendered}{quote_char}'
 
 
+_HAS_RENDER_CHARS_PAT = re.compile(r'{[{%]')
+
+
 def get_rendered(
     string: str,
     ctx: Dict[str, Any],
     node=None,
     capture_macros: bool = False,
     native: bool = False,
 ) -> str:
+    # performance optimization: if there are no jinja control characters in the
+    # string, we can just return the input. Fall back to jinja if the type is
+    # not a string or if native rendering is enabled (so '1' -> 1, etc...)
+    # If this is desirable in the native env as well, we could handle the
+    # native=True case by passing the input string to ast.literal_eval, like
+    # the native renderer does.
+    if (
```

Comment:
the one implication of this is that invalid jinja will return plaintext instead of some sort of compiler error, eg:

```
{ { config(....) }}
```

I think that's ok, but we should keep this in mind if anyone write in with a support question about something like this :)

Reply:
Well, one thing we could do there is also search for `[%}]}`. That's a pretty minor fix that might actually capture all cases that jinja would error about - I think jinja would just silently pass `{ { config(...) } }` as text, right?

Reply:
yep - i'd be in favor of also searching for `[%}]}`. It's a pretty minor failure mode, but I do think it's a good and positive change to make :) 

I'm not at all concerned about
```
{ { config() } }
```

that is plaintext as far as i'm concerned